tests/unit/client/data-access/MessagesRepository.spec.ts

```diff
@@ -0,0 +1,26 @@
+import MessagesRepository from '@/client/data-access/MessagesRepository';
+import MockMwMessage from '@/mock-data/MockMwMessage';
+import { MwMessage } from '@/client/mediawiki/MwWindow';
+
+describe( 'MessagesRepository', () => {
+	it( 'get mw.message and the messages keys to build a message collection', () => {
+		const keys = [ 'msg1', 'msg2', 'msg3' ];
+		const messagePairs = {
+			msg1: 'Nachricht1',
+			msg2: 'noch ne Nachricht',
+			msg3: 'benachrichtigt',
+		} as any;
+
+		const messages: MwMessage = ( key: string, ...params: string[] ) => new MockMwMessage( messagePairs[ key ] );
```

Comment:
IMO we don't really need to use the `MockMwMessage` class here. We can create a test implementation here, e.g. like so

```ts
		const messagePairs: { [key: string]: string } = {
			msg1: 'Nachricht1',
			msg2: 'noch ne Nachricht',
			msg3: 'benachrichtigt',
		};

		const messages = ( key: string ) => {
			return { text() { return messagePairs[ key ]; } };
		};
```

Reply:
done